File: mario.py
# Jason Berinsky
# CS 50 Problem Set 6
# Creates a pyramid of # blocks of an inputted size

from cs50 import get_int

# Prints the blocks


def blocks(n):
    for j in range(n):
        print("#", end="")

# Prints the spaces leading up to the first block


def spaces(n):
    for k in range(1, n):
        print(" ", end="")


# Gets an input value between 1 and 8 inclusive
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break


# Creates the block pyramid
level = height
for i in range(1, height + 1):
    spaces(level)
    blocks(i)
    print("  ", end="")
    blocks(i)
    print()
    level = level - 1

File: mario.c
// Jason Berinsky
// 2 February 2019
// CS50 Problem Set 1
// Displays a block pyramid of a specified height between 1 and 8
#include <cs50.h>
#include <stdio.h>

void blocks(int);
void spaces(int);

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");  
    }
    while (height < 1 || height > 8);
    int level = height;
    for (int i = 1; i <= height; i++)
    {
        spaces(level);
        blocks(i);
        printf("  ");
        blocks(i);
        printf("\n");
        level--;
    }
}

// Prints an inputted number of blocks (#)
void blocks(i)
{
    for (int j = 0; j < i; j++)
    {
        printf("#");
    }
}

// Prints an inputted number of spaces
void spaces(i)
{
    for (int j = 1; j < i; j++)
    {
        printf(" ");
    }
}
